src/audio/eq_fir/eq_fir_ipc4.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <ipc/stream.h>
+#include <sof/audio/component.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/list.h>
+#include <sof/trace/trace.h>
+#include <errno.h>
+
+#include "eq_fir.h"
+
+LOG_MODULE_DECLARE(eq_fir, CONFIG_SOF_LOG_LEVEL);
+
+int set_fir_func(struct processing_module *mod, enum sof_ipc_frame fmt)
```

Comment:
I haven't earlier needed different processing function select for ipc versions. Are the source and sink params set from init() IPC before selecting the processing function?